src/CalcManager/CEngine/scicomm.cpp

```diff
@@ -789,6 +758,45 @@ void CCalcEngine::ProcessCommandWorker(OpCode wParam)
 
 }
 
+// Helper function to resolve one item on the precedence stack.
+void CCalcEngine::ResolveHighestPrecedenceOperation()
+{
+    if (m_nOpCode) /* Is there a valid operation around?        */
+    {
+        /* If this is the first EQU in a string, set m_holdVal=m_currentVal */
+        /* Otherwise let m_currentVal=m_holdVal.  This keeps m_currentVal constant */
+        /* through all EQUs in a row.                     */
+        if (m_bNoPrevEqu)
+        {
+            m_holdVal = m_currentVal;
+        }
+        else
+        {
+            m_currentVal = m_holdVal;
+            DisplayNum(); // to update the m_numberString
+            m_HistoryCollector.AddBinOpToHistory(m_nOpCode, false);
+            m_HistoryCollector.AddOpndToHistory(m_numberString, m_currentVal); // Adding the repeated last op to history
+        }
+
+        // Do the current or last operation.
+        m_currentVal = DoOperation(m_nOpCode, m_currentVal, m_lastVal);
+        m_nPrevOpCode = m_nOpCode;
+        m_lastVal = m_currentVal;
+
+        /* Check for errors.  If this wasn't done, DisplayNum */
+        /* would immediately overwrite any error message.     */
+        if (!m_bError)
+            DisplayNum();
```

Comment:
I realize that you tried to just move the code verbatim to this helper function, but it would be great to clean it up a little bit since we're touching it.  Here's a good opportunity to wrap the conditional in brackets, since it's generally asking for trouble to leave single-line conditionals like this.

Optional: simplify a number of the comments from being `/*   */` with lots of unnecessary spaces trying to (unsuccessfully) keep alignment to just using the `//` syntax.